swagger.yml

```diff
@@ -0,0 +1,90 @@
+# swagger.yml
+# navigate to localhost:xxxx/api/ui to view API documentation
+
+# title: Title included in the Connexion-generated UI system
+# description: Description of what the API provides or is about
+# version: Version value for the API
+openapi: 3.0.0
+info:
+  title: "QueueTime Flask REST API"
+  description: "QueueTime API for finding wait times of various points of interests at McMaster University"
+  version: "1.0.0"
+
+# By providing "/api" as the value of url, you’ll be able to access all of your API paths relative to http://localhost:xxxx/api.
+servers:
+  - url: "/api"
+
+components:
+  schemas:
+    POI:
+      type: "object"
+      required:
+        - _id
+      properties:
+        _id:
+          type: "string"
+        name:
+          type: "string"
+        address:
+          type: "string"
+        class:
+          type: "string"
+        hours_of_operation:
+          type: "object"
+        image_url:
+          type: "string"
+        location:
+          type: "object"
+        type:
+          type: "string"
+  parameters:
+    poi_name:
+      name: "poi_name"
+      description: "Point of interest name of the POI to get"
+      in: path
+      required: True
+      schema:
+        type: "string"
+
+# /endpoint_name: The relative URL of your API endpoint
+# get: The HTTP method that this URL endpoint will respond to
+# operationId: The Python function that’ll respond to the request
+# tags: The tags assigned to this endpoint, which allow you to group the operations in the UI
+# summary: The UI display text for this endpoint
+# responses: The status codes that the endpoint responds with
+paths:
+  # POI API endpoints
+  /places/list:
+    get:
+      x-openapi-router-controller: app.poi_api.poi.POI
+      operationId: list_POI
+      tags:
+        - POI
+      summary: "Read the list of POI"
+      responses:
+        "200":
+          description: "Successfully read POI list"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/POI'
+  /places/list/{poi_name}:
+    get:
+      x-openapi-router-controller: app.poi_api.poi
+      operationId: get_details_for_POI
+      tags:
+        - POI
+      summary: "Read one POI"
+      parameters:
+        - $ref: "#/components/parameters/poi_name"
+      responses:
+        "200":
+          description: "Successfully read the POI"
+          content:
+            application/json:
+              schema:
+                type: array

```

Comment:
Is this returning a single POI or multiple?

Reply:
Oh nice catch, this is just single, I will change to object